app/tests/test_search_endpoint.py

```diff
@@ -17,6 +20,153 @@
+        response = self.client.get(previous_link['href'])
+        self.assertStatusCode(200, response)
+        json_data_previous = response.json()
+        self.assertEqual(len(json_data_previous['features']), limit)
+
+        # make sure the previous data is identical to the first page
+        self.assertEqual(
+            json_data_previous['features'],
+            json_data['features'],
+            msg='previous page should be the same as the first one'
+        )
+
+    def test_post_pagination(self):
+        limit = 1
+        query = {"query": {"title": {"startsWith": self.title_for_query}}, "limit": limit}
+        response = self.client.post(self.path, data=query, content_type="application/json")
+        self.assertStatusCode(200, response)
+        json_data = response.json()
+        self.assertEqual(len(json_data['features']), limit)
+        # get the next link
+        next_link = get_link(json_data['links'], 'next')
+        self.assertIsNotNone(next_link, msg='No next link found')
+        self.assertEqual(next_link.get('method', 'POST'), 'POST')
+        self.assertIn('body', next_link)
+        self.assertIn('merge', next_link)
+
+        # Get the next page
+        query_next = query.copy()
+        if next_link['merge'] and next_link['body']:
+            query_next.update(next_link['body'])
+        response = self.client.post(
+            next_link['href'], data=query_next, content_type="application/json"
+        )
+        self.assertStatusCode(200, response)
+        json_data_next = response.json()
+        self.assertEqual(len(json_data_next['features']), limit)
+
+        # make sure the next page is different than the original
+        self.assertNotEqual(
+            json_data['features'],
+            json_data_next['features'],
+            msg='Next page should not be the same as the first one'
+        )
+
+        # get the previous link
+        previous_link = get_link(json_data_next['links'], 'previous')
+        self.assertIsNotNone(previous_link, msg='No previous link found')
+        self.assertEqual(previous_link.get('method', 'POST'), 'POST')
+        self.assertIn('body', previous_link)
+        self.assertIn('merge', previous_link)
+
+        # Get the previous page
+        query_previous = query.copy()
+        if previous_link['merge'] and previous_link['body']:
+            query_previous.update(previous_link['body'])
+        response = self.client.post(
+            previous_link['href'], data=query_previous, content_type="application/json"
+        )
+        self.assertStatusCode(200, response)
+        json_data_previous = response.json()
+        self.assertEqual(len(json_data_previous['features']), limit)
+
+        # make sure the previous data is identical to the first page
+        self.assertEqual(
+            json_data_previous['features'],
+            json_data['features'],
+            msg='previous page should be the same as the first one'
+        )
+
```

Comment:
Probably it might make sense to also implement this in `check_next_and_previous_link()` in the e2e tests in here:
https://github.com/geoadmin/infra-e2e-tests/blob/master/tests/test_service_stac/test_pagination.py